### PR TITLE
Create story promo list component

### LINKED ISSUE #477: Create story promo list
An a11y criteria is to have story promos in a list when there is more than 1 item

**Describe the solution you'd like**
Create an unordered list and list item component with no styling except a separator line

```
import { unorderedList, listItem } from '@bbc/psammead-story-promo-list'
```
The list should have zero styling, padding or margins except for the separator line, which will be a border bottom. 

The separator shouldn't appear at the bottom of all list items or at the top.

UX spec for separator line:

- Height: 1px
- Colour: `C_LUNAR` from `@bbc/psammead-styles/colours`
- Spacing:
  - `0` - `600px`:
    - Above: 16px - `GEL_SPACING_DBL` from `@bbc/gel-foundations`
    - Below: 8px - `GEL_SPACING` from `@bbc/gel-foundations`
  - +`600px`:
    - Above: 16px - `GEL_SPACING_DBL` from `@bbc/gel-foundations`
    - Below: 16px - `GEL_SPACING_DBL` from `@bbc/gel-foundations`
  - +`1008px`:
    - Above: 24px - `GEL_SPACING_TRPL` from `@bbc/gel-foundations`
    - Below: 24px - `GEL_SPACING_TRPL` from `@bbc/gel-foundations`

Developer note, Role `list` and role `listitem` will need to be applied to the corresponding list items due to a VoiceOver bug to reinstate the list semantics

**Testing notes**
[Tester to complete]

Dev insight: Snapshots and storybook
Storybook should include multiple stories in a list, built up.
UX will provide information about the stories to show in storybook

**Additional context**
[Screenreader UX](https://docs.google.com/document/d/1dBN1hAL6VrXWnIEDgYyR_MJP6UgQFnW_KiqcTLUeGUU/edit#)

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #477

**Overall change:** 
Implement an unordered list and a list item component with no styling except a separator line to be used with `StoryPromos`.

**Code changes:**
- Create `StoryPromoUl` component
- Create `StoryPromoLi` component

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-story-promo-list/src/index.stories.jsx
Comment (dr3) [inline packages/components/psammead-story-promo-list/src/index.stories.jsx]: As its not in prod, nanoid is fine here, but it needs to be inc in the package.json :)

Comment (dr3) [reply packages/components/psammead-story-promo-list/src/index.stories.jsx]: for context, we are moving away from nanoid due to the generated ID being different in simorgh server/client renders, but thats not a problem here :)

File: packages/components/psammead-story-promo-list/README.md
Comment (dr3) [inline packages/components/psammead-story-promo-list/README.md:4]: As this is implemented, the top and button padding of the whole `ul` is determined by the separator spacing logic. Can you make it so the top and buttom padding is 0 please (i.e no top padding on first `li` and no bottom padding on the last).
It means that the components around it, or the page wrapper, can determine its surrounding padding

**Review by dr3 — CHANGES_REQUESTED** [review]

File: packages/components/psammead-story-promo-list/README.md
Comment (dr3) [inline packages/components/psammead-story-promo-list/README.md]: `children` can be listed here

File: packages/components/psammead-story-promo-list/src/index.jsx
Comment (dr3) [inline packages/components/psammead-story-promo-list/src/index.jsx]: Just realised we should do the spec
>Role list and role listitem will need to be applied to the corresponding list items due to a VoiceOver bug to reinstate the list semantics

As this will be the case for every use of this component, we should enforce it here. To do this, rather than exporting a styled component, we should export a react component.

This can be done like

```
const StyledUnorderedList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const StoryPromoUl = ({ children, ...props }) => {
  <StyledUnorderedList role="list" {...props} />
    {children}
  </StyledUnorderedList>
}
```

File: packages/components/psammead-story-promo-list/testHelpers/fixtureData.js
Comment (dr3) [inline packages/components/psammead-story-promo-list/testHelpers/fixtureData.js:1]: Can you replace this data with something more representative of stories please :) Realistic length headlines and summaries make it easier to judge the read world look on an index page

Also, we should ideally use bbc owned/free use images

File: packages/components/psammead-story-promo-list/src/index.stories.jsx
Comment (DenisHdz) [reply packages/components/psammead-story-promo-list/src/index.stories.jsx]: Although `nanoid` is fine, shall I use another `key`?

Comment (dr3) [reply packages/components/psammead-story-promo-list/src/index.stories.jsx]: You could use something like the headline yeah

File: packages/components/psammead-story-promo-list/src/index.jsx
Comment (DenisHdz) [reply packages/components/psammead-story-promo-list/src/index.jsx]: May I ask why do we need `...props` here?

Comment (dr3) [reply packages/components/psammead-story-promo-list/src/index.jsx]: Just to allow the user to add any extra props if they wanted, gives more freedom really

File: packages/components/psammead-story-promo-list/src/index.jsx
Comment (dr3) [inline packages/components/psammead-story-promo-list/src/index.jsx:43]: Hey, soz can you do the same react component design with `StoryPromoLi` with role `listitem` (i think(double check a11y spec))

**Review by dr3 — CHANGES_REQUESTED** [review]

File: packages/components/psammead-story-promo-list/src/index.jsx
Comment (dr3) [inline packages/components/psammead-story-promo-list/src/index.jsx:19]: Nice to have clear separation imo
```suggestion

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
```

File: packages/components/psammead-story-promo-list/src/index.jsx
Comment (dr3) [inline packages/components/psammead-story-promo-list/src/index.jsx:23]:
```suggestion

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
```

File: packages/components/psammead-story-promo-list/testHelpers/fixtureData.js
Comment (dr3) [inline packages/components/psammead-story-promo-list/testHelpers/fixtureData.js]: You dont appear to be using the time value here.
Can you add a key for the date format `12 March 2019` and use both please?

**Review by dr3 — CHANGES_REQUESTED** [review]

**Review by dr3 — DISMISSED** [review]
lgtm

**Review by j-pendlebury — DISMISSED** [review]
branch is out of date but happy to approve once that's fixed :)

**Comment by david-boydell** [comment]
Looks good to me. Due to network issues will have to check cross device over the weekend.

File: packages/components/psammead-story-promo-list/README.md
Comment (sareh) [inline packages/components/psammead-story-promo-list/README.md]: You've added roles to the list and listitem elements, which is great. This should be explained here in the Accessibility notes, so that users of this package understand the reasoning for this. 

Since this is a new component, it should go through an accessibility swarm. This is to check this assistive technologies to ensure that it is working as expected, prior to merging. For example, that the list and list  items are being read out correctly by screenreaders. 
This swarm should be done prior to merge, since once merged, it will be published to npm and be available for others to use.


Here are instructions for how to use the various assistive technologies that we support: https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology

If you have any questions, please let me know.

File: packages/components/psammead-story-promo-list/src/index.test.jsx
Comment (sareh) [inline packages/components/psammead-story-promo-list/src/index.test.jsx]: The component already adds these roles, so you do not need to add it here again. 
```suggestion
    <StoryPromoUl>
```

File: packages/components/psammead-story-promo-list/src/index.test.jsx
Comment (sareh) [inline packages/components/psammead-story-promo-list/src/index.test.jsx]:
```suggestion
      <StoryPromoLi>
```

File: packages/components/psammead-story-promo-list/src/index.test.jsx
Comment (DenisHdz) [reply packages/components/psammead-story-promo-list/src/index.test.jsx]: Good spot!

**Comment by DenisHdz** [comment]
We should carry on an Accessibility Swarm on the `StoryPromoList` once the `StoryPromo` is finished and swarmed.

**Comment by dr3** [comment]
Chatting on slack, we've agreed we can merge this component, as an alpha. Can you please tag it using 
```
"publishConfig": {
  "tag": "alpha",
}
```
in the `package.json`, make this version `0.1.0-alpha.0` and add
```
# ⛔️ This is an alpha component  ⛔️
This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed.
```
to the top of the readme :)

**Comment by greenc05** [comment]
We might want to tweak our use of English slightly... "This component is currently tagged as alpha and is not suitable for production use. Following the passing of an accessibility review this component will be marked as ready for production and the alpha tag removed."

**Review by dr3 — DISMISSED** [review]
LGTM, great work :P

Weve never published anything as alpha before, lets hope [my publishing script](https://github.com/bbc/psammead/blob/latest/scripts/publish/src/publish.js#L18) works as expected :D

File: packages/components/psammead-story-promo-list/README.md
Comment (ghost) [inline packages/components/psammead-story-promo-list/README.md]:
```suggestion
The `@bbc/psammead-story-promo-list` package is a set of two components, `StoryPromoUl` and `StoryPromoLi`. They use `ul` and `li` HTML elements respectively.
```

**Review by j-pendlebury — DISMISSED** [review]
LGTM 👍

File: packages/components/psammead-story-promo-list/README.md
Comment (dr3) [reply packages/components/psammead-story-promo-list/README.md]: I dont think thats rights, `StoryPromoLi` only uses a single `li` element

**Review by dr3 — APPROVED** [review]

**Review by j-pendlebury — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by david-boydell** [comment]
LGTM, a11y to be conducted once full component is delivered

**Pull request merged**
